File: python/test/test_count.py
import ThrustRTC as trtc



op = trtc.Functor( {}, ['x'], 
'''
         return x % 100;
''')


darr = trtc.device_vector('int32_t', 2000)
trtc.Transform(trtc.DVCounter(trtc.DVInt32(0), 2000), darr, op)
print(trtc.Count(darr, trtc.DVInt32(47)))


op2 = trtc.Functor({}, ['x'],
'''
         return (x % 100)==47;
''')

trtc.Sequence(darr)
print(trtc.Count_If(darr, op2))

File: java/test/test_count.java
import java.util.*;
import JThrustRTC.*;

public class test_count
{
	public static void main(String[] args) 
	{
        int[] hin=new int[2000];
        for (int i = 0; i < 2000; i++)
            hin[i] = i % 100;

        DVVector din = new DVVector(hin);
		System.out.println(TRTC.Count(din, new DVInt32(47)));

        TRTC.Sequence(din);
        Functor op = new Functor(new String[]{ "x" }, "        return (x%100)==47;\n" );
		System.out.println(TRTC.Count_If(din, op));
	}
}
